config/initializers/devise.rb

```diff
@@ -50,7 +51,11 @@
   # session. If you need permissions, you should implement that in a before filter.
   # You can also supply a hash where the value is a boolean determining whether
   # or not authentication should be aborted when the value is not present.
-  # config.authentication_keys = [:email]
+  if Rails.configuration.x.use_ldap
+    config.authentication_keys = [:ldap_username]
```

Comment:
I tried in vain, but as I said I have little experience with that, if someone knows better...